src/cdtools/datasets/ptycho_2d_dataset.py

```diff
@@ -398,3 +398,47 @@ def downsample(self, factor=2):
                 self.background.unsqueeze(0).unsqueeze(0),
                 factor,
                 divisor_override=1)[0,0]
+
+
+    def crop_2D_translations(self, roi):
+        """Shrinks the range of translation positions that are analyzed
+        
+        This deletes all diffraction patterns associated with x- and 
+        y-translations that lie outside of a specified rectangular
+        region of interest. In essence, this operation crops the "relative 
+        displacement map" (shown in self.inspect()) down to the region of 
+        interest.
+
+        Parameters:
+        ----------
+        roi : tuple(float, float, float, float)
+            The translation-x and -y coordinates that define the rectangular 
+            region of interest as (in units of meters)
+            ('left-most x', 'right-most x', 'top-most y', 'bottom-most y').
+            Note that "left" and "up" are defined to be the positive-most
+            values of x and y, respectively. This is to have consistency with
+            the "relative displacement map" in self.inspect().
+        """
+        
+        # Convert the tuples to torch tensors
+        x_left, x_right, y_top, y_bottom = t.tensor(roi, dtype=t.float16)
```

Comment:
what is t.tensor accomplishing here? Serious question, I'm not sure if it is needed or not 😅. It might not be needed, because you then specifically pull out the four values.

I do also worry about the explicit casting to a float16, if there was no specific reason to do it, I think it's best to preserve the original accuracy.

Reply:
Conversion to a torch tensor is not required here; this was from a misunderstanding I had regarding logical indexing with torch tensors.

Rewriting this bit to just pull out four values.